stats/registry_test.go

```diff
@@ -0,0 +1,50 @@
+package stats_test
+
+import (
+	"testing"
+
+	"github.com/influxdata/influxdb/stats"
+)
+
+func TestEmptyStatistics(t *testing.T) {
+	found := stats.Collect(stats.Root.Open(), true)
+
+	if length := len(found); length != 0 {
+		t.Fatalf("non empty initial state. got %d, expected: %d", length, 0)
+	}
+}
+
+// Test that we can create one statistic and that it disappears after it is deleted twice.
+func TestOneStatistic(t *testing.T) {
+
+	found := stats.Collect(stats.Root.Open(), true)
+
+	foo := stats.Root.
+		NewBuilder("foo", map[string]string{"tag": "T"}).
+		MustBuild().
+		Open()
+	defer func() {
+		if foo != nil {
```

Comment:
If I don't use defer at line 26, I need to issue a foo.Close() before every t.Fatalf() otherwise a resource will leak which may lead to other test failures. However, I don't mind making that change.